docs/src/modules/components/MuiProductSelector.tsx

```diff
@@ -47,44 +50,58 @@ interface ProductItemProps {
   name: string;
 }
 
-function ProductItem({ active, chip, description, href, icon, name }: ProductItemProps) {
+function ProductItem({
+  active,
+  chip,
+  description,
+  href,
+  icon,
+  name,
+  sx = [],
+  ...other
+}: ProductItemProps) {
   return (
-    <Box
+    <MenuItem
```

Comment:
@alexfauquette Loving the changes here (and the keyboard nav fix 🙏), but I don't think we can have `a` tags as children of a `ul`. I mean, the browser still renders it, but it seems like [it's invalid HTML](https://stackoverflow.com/questions/12242041/allowed-child-elements-of-ul)? That is what's happening now if the `MenuItem` (rendering `a`) is a child of `MuiProductSelector` (Menu List, rendering `ul`).

<img width="400" alt="Screenshot 2024-06-12 at 10 17 25" src="https://github.com/mui/material-ui/assets/67129314/72efc654-03c6-46fb-9697-f9d580681584">